specification/batch/data-plane/Microsoft.Batch/2017-09-01.6.0/BatchService.json

```diff
@@ -12605,17 +12719,88 @@
+        },
+        "dataDisks": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DataDisk"
+          },
+          "title": "The configuration for data disks attached to the comptue nodes in the pool.",
+          "description": "This property must be specified if the compute nodes in the pool need to have empty data disks attached to them. This cannot be updated."
+        },
+        "licenseType": {
+          "type": "string",
+          "title": "If the the image or disk that is being used was licensed on-premises.",
+          "description": "This only applies to images that contain the Windows operating system. Values are:\n\n Windows_Server - The on-premises license is for Windows Server.\n Windows_Client - The on-premises license is for Windows Client."
+        },
+        "containerConfiguration": {
+          "$ref": "#/definitions/ContainerConfiguration",
+          "title": "The container configuration for the pool.",
+          "description": "If this is specified, then all regular tasks and job manager tasks run on this pool must specify the containerSettings property."
         }
       },
-      "required": [ "nodeAgentSKUId" ],
+      "required": [ "nodeAgentSKUId", "imageReference" ],
       "title": "The configuration for compute nodes in a pool based on the Azure Virtual Machines infrastructure."
     },
+    "ContainerRegistry": {
+      "properties": {
+        "registryServer": {
+          "type": "string",
+          "title": "The registry server url.",
+          "description": "If omitted, the default is: \"docker.io\"."
+        },
+        "username": {
+          "type": "string",
+          "x-ms-client-name": "userName",
+          "title": "The user name to login to the registry server."
+        },
+        "password": {
+          "type": "string",
+          "title": "The password to login to the registry server."
+        }
+      },
+      "required": [ "username", "password" ],
+      "title": "Private container registry details."
+    },
+    "ContainerConfiguration": {
+      "properties": {
+        "type": {
+          "type": "string",
+          "title": "Which container technology will be used.",
+          "description": "Values are:\n\n docker - The container will use Docker.",
+          "enum": [
+            "docker"
+          ],
+          "x-ms-enum": {
+            "name": "ContainerType",
+            "modelAsString": false
+          }
+        },
+        "containerImageNames": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "title": "The collection of container image names.",
+          "description": "This is the full image reference, as would be specified to \"docker pull\". An image will be sourced from the default Docker registry unless the image is fully qualified with an alternative registry."
```

Comment:
Bit confused how this interacts with the "containerRegistries" list.  So if I am using a custom registry then I need to fully qualify the image reference here, but I still need to list it in containerRegistries, or only if I need to provide login credentials?  If the latter then maybe we need to clarify that in the docs.

Reply:
If containerRegistries is about a map from registry URLs to private login credentials then we should consider not defaulting the URL in the ContainerRegistry type - that will make the mapping more explicit.

Reply:
For your first comment, it is only required if you need login credentials. I've clarified that.

For the second comment, Xing says that the intent is to match `docker login` behavior here, and that behavior says that if just a username/password are provided for the registry, the URL defaults to docker.io (which is what we are doing now too).

Reply:
Okay, matching community precedent works for me.  Thanks for explaining.